internal/test/e2e/utils_test.go

```diff
@@ -781,3 +781,44 @@ func refill(httpClient *http.Client) error {
 	}
 	return nil
 }
+
+type pendingTxResponse struct {
+	PendingTxs []pendingTx `json:"pendingTxs"`
+}
+
+type pendingTx struct {
+	ArkTxid             string   `json:"arkTxid"`
+	FinalArkTx          string   `json:"finalArkTx"`
+	SignedCheckpointTxs []string `json:"signedCheckpointTxs"`
+}
+
+func getPendingTxByTxid(httpClient *http.Client, txid string) (*pendingTxResponse, error) {
```

Comment:
we're gonna open pr on go-sdk for u so that we can drop this. Will ping u as soon as it's done